include/domain_bridge/domain_bridge.hpp

```diff
@@ -142,8 +156,12 @@ class DomainBridge
 
 private:
   std::unique_ptr<DomainBridgeImpl> impl_;
+  std::unique_ptr<ServiceBridgeImpl> service_bridge_impl_;
```

Comment:
What's the reason (or trade-off) for adding a second pimpl versus bundling the implementation into `DomainBridgeImpl`? For example, `DomainBridgeImpl` could itself have a `ServiceBridgeImpl` object.

It seems like it would be better to include `service_bridge_impl.inc` (or just rename to `service_bridge_impl.hpp`?) directly in `domain_bridge.cpp`. Then we don't have to worry about polluting the namespace here.

Reply:
> It seems like it would be better to include service_bridge_impl.inc directly in domain_bridge.cpp

We cannot do that because this is a template method, so the definition needs to live in a header file.

> or just rename to service_bridge_impl.hpp?

I followed the google c++ style guide [suggestion](https://google.github.io/styleguide/cppguide.html#Self_contained_Headers):

```
Non-header files that are meant for inclusion should end in .inc and be used sparingly.
```

> What's the reason (or trade-off) for adding a second pimpl versus bundling the implementation into DomainBridgeImpl? For example, DomainBridgeImpl could itself have a ServiceBridgeImpl object.

Template methods make pimpl really complicated, as the definition of the method needs to be available in a header file.
If I move the `DomainBridgeImpl` definition to the header, then I break the idea of `pimpl`, because you cannot guarantee ABI stability anymore (when the "impl" definition is available in a header file and some of the methods definitions too, the compiler can start doing inlining and it's hard to guarantee ABI stability).

The only way to avoid this issues is to add some "type erased" methods to `DomainBridgeImpl` and use them from the template method definition. I think I can manage to do that.

Reply:
ah, I see the issues now. Let's leave it as-is.

Reply:
I refactored a bit in https://github.com/ros2/domain_bridge/pull/26/commits/ec4f8862d05a871beab07cfaa8257a8349b977c6, now there's only one `impl_` member.